subprojects/base-services/src/test/groovy/org/gradle/internal/operations/BuildOperationIdentifierRegistryTest.groovy

```diff
@@ -0,0 +1,21 @@
+package org.gradle.internal.operations
+
+import spock.lang.Specification
+import spock.lang.Subject
+
+@Subject(BuildOperationIdentifierRegistry)
+class BuildOperationIdentifierRegistryTest extends Specification {
+    def "able to set and retrieve current operation across threads"() {
```

Comment:
should have `expect: BuildOperationIdentifierRegistry.getCurrentOperationIdentifier() == null` to verify the initial state.